lam4-backend/src/Lam4/Expr/Printer.hs

```diff
@@ -266,12 +281,23 @@ instance Print Expr where
     Let decl expr -> prPrec i 0 (concatD [doc (showString "LET"), doc (showString "{"), prt 0 decl, doc (showString "}"), doc (showString "IN"), doc (showString "{"), prt 0 expr, doc (showString "}")])
     StatementBlock (st :| sts) -> prPrec i 0 (concatD [prt 0 (st:sts)])
     NormIsInfringed name -> prPrec i 0 (concatD [prt 0 name, doc (showString "IS_INFRINGED")])
-    Predicate metadata names expr ->
+    Predicate metadata [funname] expr ->
       prPrec i 0 (concatD [
           prt 0 metadata
-        , doc (showString "GIVEN ()")
         , doc (showString "\nDECIDE")
-        , prt 0 names
+        , prt 0 funname
+        , doc (showString "\nIF")
+        , prt 0 expr
+        ])
+    Predicate metadata (funname:args) expr ->
+      prPrec i 0 (concatD [
+          prt 0 metadata
+        , doc (showString "GIVEN")
+        , doc (showString "(")
+        , prt 0 (fmap (\a -> mkFunArg a.name) args) -- custom newtype to print args in a specific way
+        , doc (showString ")")
+        , doc (showString "\nDECIDE")
+        , prt 0 funname
```

Comment:
Another example from the file as in the comment above: [examples/infix_pred_app.l4](https://github.com/smucclaw/lam4/blob/main/examples/infix_pred_app.l4).

* Original:
    ```
    GIVEN (person: Person)
    DECIDE `is fun`
    IF True
    ```
* CST:
    ```
    Rec
      "is fun"
      _8
      (Predicate
        RuleMetadata
          { originalRuleRef = Nothing
          , transparency = Transparent
          , description = Nothing
          }
        [ "person" _9 ]
        (Lit (BoolLit True)))
     ```
With the current line 300, this is prettyprinted as follows:
```
GIVEN (person : TypeMissing)
DECIDE `is fun`
IF True
```
so it happens to work in this case, except for the TypeMissing part. But for the other cases where the predicate is actually applied to its argument, it's not working.

(pinging @ym-han for this and the comment on lines 138-143)